radix-engine-interface/src/model/resource/non_fungible_id.rs

```diff
@@ -43,10 +42,89 @@ impl NonFungibleId {
             NonFungibleId::String(..) => NonFungibleIdType::String,
             NonFungibleId::U32(..) => NonFungibleIdType::U32,
             NonFungibleId::U64(..) => NonFungibleIdType::U64,
-            NonFungibleId::Decimal(..) => NonFungibleIdType::Decimal,
             NonFungibleId::UUID(..) => NonFungibleIdType::UUID,
         }
     }
+
+    /// Returns string representation of non-fungible ID value for transaction manifest.
+    pub fn to_manifest_string(&self) -> String {
+        match self {
+            NonFungibleId::Bytes(b) => format!("Bytes(\"{}\")", hex::encode(b)),
+            NonFungibleId::String(s) => format!("\"{}\"", s),
+            NonFungibleId::U32(n) => format!("{}u32", n),
+            NonFungibleId::U64(n) => format!("{}u64", n),
+            NonFungibleId::UUID(u) => format!("{}u128", u),
+        }
+    }
+
+    /// Converts transaction manifest representation string to non-fungible ID.
+    pub fn try_from_manifest_string(s: &str) -> Result<Self, ParseNonFungibleIdError> {
+        let s = s.trim();
+        if s.len() > 9 && s.starts_with("Bytes(\"") && s.ends_with("\")") {
+            Ok(NonFungibleId::Bytes(
+                hex::decode(&s[7..s.len() - 2])
+                    .map_err(|_| ParseNonFungibleIdError::InvalidValue)?,
+            ))
+        } else if s.len() > 4 && s.ends_with("u128") {
+            Ok(NonFungibleId::UUID(
+                s[0..s.len() - 4]
+                    .parse::<u128>()
+                    .map_err(|_| ParseNonFungibleIdError::InvalidValue)?,
+            ))
+        } else if s.len() > 3 && s.ends_with("u64") {
+            Ok(NonFungibleId::U64(
+                s[0..s.len() - 3]
+                    .parse::<u64>()
+                    .map_err(|_| ParseNonFungibleIdError::InvalidValue)?,
+            ))
+        } else if s.len() > 3 && s.ends_with("u32") {
+            Ok(NonFungibleId::U32(
+                s[0..s.len() - 3]
+                    .parse::<u32>()
+                    .map_err(|_| ParseNonFungibleIdError::InvalidValue)?,
+            ))
+        } else if s.len() > 2 && s.starts_with("\"") && s.ends_with("\"") {
+            Ok(NonFungibleId::String(s[1..s.len() - 1].to_string()))
+        } else {
+            Err(ParseNonFungibleIdError::InvalidValue)
+        }
```

Comment:
This is a very fragile implementation. Can be wrong for many edge cases, e.g. `Bytes( "deadbeef" )` and `"abc"d"`. We should probabaly use manifest parser to do the this job.

In the interest of time, I'd suggest we add a `TODO` note and merge it as-is.

Reply:
done